fluss-common/src/main/java/org/apache/fluss/utils/json/DataTypeJsonSerde.java

```diff
@@ -285,7 +285,8 @@ private static DataType deserializeMap(JsonNode dataTypeNode) {
         final DataType keyType = DataTypeJsonSerde.INSTANCE.deserialize(keyNode);
         final JsonNode valueNode = dataTypeNode.get(FIELD_NAME_VALUE_TYPE);
         final DataType valueType = DataTypeJsonSerde.INSTANCE.deserialize(valueNode);
-        return new MapType(keyType, valueType);
+        final DataType nonNullableKeyType = keyType.isNullable() ? keyType.copy(false) : keyType;
```

Comment:
do not transform the data type during deserialize, this is very hack. If the key shouldn't be nullable, then this should be guaranteed by `MapType` constructor, not the type serialization/deserialition phase. Serialization/deserialition should just focus on how to read and write types from JSON.